### PR TITLE
kubeadm: update generated files

### PR DESCRIPTION
**Pull request by nikhita**
Verify is currently broken on master. This PR fixes it.

See https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/65811/pull-kubernetes-verify/97686/ for an example error.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nikhita** [comment]
```
 diff --git a/staging/src/k8s.io/api/imagepolicy/v1alpha1/types_swagger_doc_generated.go b/staging/src/k8s.io/api/imagepolicy/v1alpha1/types_swagger_doc_generated.go
  index b9a67a9bfd..52c1ce9447 100644
  --- a/staging/src/k8s.io/api/imagepolicy/v1alpha1/types_swagger_doc_generated.go
  +++ b/staging/src/k8s.io/api/imagepolicy/v1alpha1/types_swagger_doc_generated.go
  @@ -58,9 +58,9 @@ func (ImageReviewSpec) SwaggerDoc() map[string]string {
   }
   
   var map_ImageReviewStatus = map[string]string{
  -	"":        "ImageReviewStatus is the result of the token authentication request.",
  -	"allowed": "Allowed indicates that all images were allowed to be run.",
  -	"reason":  "Reason should be empty unless Allowed is false in which case it may contain a short description of what is wrong.  Kubernetes may truncate excessively long errors when displaying to the user.",
  +	"":            "ImageReviewStatus is the result of the token authentication request.",
  +	"allowed":     "Allowed indicates that all images were allowed to be run.",
  +	"reason":      "Reason should be empty unless Allowed is false in which case it may contain a short description of what is wrong.  Kubernetes may truncate excessively long errors when displaying to the user.",
   	"annotations": "Annotations should be added to the annotations of the pod. They may be used to later find pods, e.g. ones that require audits. Annotations may only be set in case 'Allowed' is True.  They should be match the pattern '*.image-policy.k8s.io/*'.",
   }
```

I tried running `hack/update-generated-swagger-docs.sh` but it didn't generate any changes. :thinking:

**Comment by nikhita** [comment]
Verify is green on this PR.

**Comment by nikhita** [comment]
/hold

running the script again on master does not generate this change; looks like a bug in the verify script (?)

**Comment by nikhita** [comment]
Maybe related: https://github.com/kubernetes/kubernetes/pull/54372.

**Comment by sttts** [comment]
My theory:

I believe our code generation check in CI is flawed: when the branch is checked out, the files get the mtime from the time of checkout (not from commit dates or something like that; also files in git trees have no dates). If you run `make`, it will invoke `Makefile.generated_files` which will do a go import analysis with go2make. Then make will compare mtimes of all files. Depending on checkout order of the conversion-gen files, they might be older than generated code files.

Now in the case of @luxas' kubeadm v1alpha3 conversion code file, he probably started working on that before my https://github.com/kubernetes/kubernetes/pull/65645 was merged. So his generated code was old. But the makefile based re-generation mechanism was not triggered.

I have not investigated how we actually merge tested PRs with master. I would expect that we checkout master and then merge a tested PR into it. Added files from the PR will then automatically be newer than all files on master. So code-gen changes on master will never trigger the regeneration.

@thockin @cblecker @luxas does this ^^ handwaving make any sense? Do we have a CI problem here?

**Comment by sttts** [comment]
/lgtm
/hold cancel
/approve

To unblock master.

**Comment by sttts** [comment]
/assign @luxas 

For approval of the kubeadm changes.

**Review by luxas — APPROVED** [review]
/approve
Thanks @nikhita!
@sttts yes, something's wrong in our CI testing that made my PR with the "old" way of auto-gen'd stuff mergeable...

**Comment by nikhita** [comment]
cc @fejta @BenTheElder

**Comment by luxas** [comment]
Yeah, something's strange here in that my PR could merge (https://github.com/kubernetes/kubernetes/pull/65629). Even more strange is that deepcopy locally was generated correctly using the new `corev1`, but conversion was still left on `core_v1` :thinking:

**Comment by BenTheElder** [comment]
> I have not investigated how we actually merge tested PRs with master. I would expect that we checkout master and then merge a tested PR into it. Added files from the PR will then automatically be newer than all files on master. So code-gen changes on master will never trigger the regeneration.

Yeah pretty much. Depending on mtime is not great. :confused: 
We could probably rig up a separate check for the generated files that does not merge first... @stevekuznetsov @cjwagner do the podutils support this?

**Comment by BenTheElder** [comment]
This seems suspect: https://github.com/kubernetes/kubernetes/pull/54372/commits/cb1332db06ad8283bdf45fb693659fc668f2dd3d
I don't think we're even verifying generated files at all now?

**Comment by cblecker** [comment]
yeah, I can't see anything that's verifying generated files at all.

Opened #65882.

**Pull request merged**
